libraries/stdlib/src/kotlin/collections/Maps.kt

```diff
@@ -483,6 +483,21 @@ public inline fun <K, V> Map<out K, V>.filterValues(predicate: (V) -> Boolean):
     return result
 }
 
+/**
+ * Returns a map containing all key-value pairs with nonnull values.
+ *
+ * The returned map preserves the entry iteration order of the original map.
+ * @sample samples.collections.Maps.Filtering.filterValuesNotNull
+ */
+public fun <K, V> Map<out K, V?>.filterValuesNotNull(): Map<K, V> {
```

Comment:
If I am not mistaken, you could use the existing function `filterTo` to shorten the code a bit.
It could look like this then
```kotlin
fun <K, V> Map<out K, V>.filterValuesNotNull(): Map<K, V> = this.filterTo(LinkedHashMap<K, V>()) { (_, value) -> value != null }
```

Reply:
That’s true. I also thought about just returning
```kotlin
this.filterValues { it != null } as Map<K, V>
```
because I’m pretty sure that’s a safe cast.
If either of those would be preferred, I’ll update the PR accordingly.